xray/commands/audit/generic/jas/applicabilityscanner.go

```diff
@@ -0,0 +1,336 @@
+		return &ExtendedScanResults{XrayResults: results, ApplicabilityScannerResults: nil, EntitledForJas: false}, nil
+	}
+	entitledForJas, err := applicabilityScanManager.Run()
+	if !entitledForJas {
+		log.Info("got not entitled error from analyzer manager")
+		return &ExtendedScanResults{XrayResults: results, ApplicabilityScannerResults: nil, EntitledForJas: false}, nil
+	}
+	if err != nil {
+		log.Info(applicabilityScanFailureMessage + err.Error())
+		return nil, err
+	}
+	applicabilityScanResults := applicabilityScanManager.getApplicabilityScanResults()
+	extendedScanResults := ExtendedScanResults{XrayResults: results, ApplicabilityScannerResults: applicabilityScanResults, EntitledForJas: true}
+	return &extendedScanResults, nil
+}
+
+func (a *ApplicabilityScanManager) shouldRun() (bool, error) {
+	analyzerManagerExist, err := a.analyzerManager.DoesAnalyzerManagerExecutableExist()
+	if err != nil {
+		return false, err
+	}
+	return analyzerManagerExist && a.resultsIncludeEligibleTechnologies() && len(a.createCveList()) > 0, nil
+}
+
+func (a *ApplicabilityScanManager) resultsIncludeEligibleTechnologies() bool {
+	for _, vuln := range a.xrayVulnerabilities {
+		for _, technology := range technologiesEligibleForApplicabilityScan {
+			if vuln.Technology == technology.ToString() {
+				return true
+			}
+		}
+	}
+	for _, violation := range a.xrayViolations {
+		for _, technology := range technologiesEligibleForApplicabilityScan {
+			if violation.Technology == technology.ToString() {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+type ApplicabilityScanManager struct {
+	applicabilityScannerResults map[string]string
+	xrayVulnerabilities         []services.Vulnerability
+	xrayViolations              []services.Violation
+	configFileName              string
+	resultsFileName             string
+	analyzerManager             AnalyzerManager
+	serverDetails               *config.ServerDetails
+}
+
+func NewApplicabilityScanManager(xrayScanResults []services.ScanResponse, dependencyTrees []*services.GraphNode,
+	serverDetails *config.ServerDetails) (*ApplicabilityScanManager, error) {
+	directDependencies := getDirectDependenciesList(dependencyTrees)
+	tempDir, err := fileutils.CreateTempDir()
+	if err != nil {
+		return nil, err
+	}
+	return &ApplicabilityScanManager{
+		applicabilityScannerResults: map[string]string{},
+		xrayVulnerabilities:         extractXrayDirectVulnerabilities(xrayScanResults, directDependencies),
+		xrayViolations:              extractXrayDirectViolations(xrayScanResults, directDependencies),
+		configFileName:              filepath.Join(tempDir, "config.yaml"),
+		resultsFileName:             filepath.Join(tempDir, "results.sarif"),
+		analyzerManager:             analyzerManagerExecuter,
+		serverDetails:               serverDetails,
+	}, nil
+}
+
+func extractXrayDirectViolations(xrayScanResults []services.ScanResponse, directDependencies []string) []services.Violation {
```

Comment:
1. Add a clear description for this function.
2. Add a unit test for it using table-testing.

The same goes for the `extractXrayDirectVulnerabilities` and `getDirectDependenciesList` functions and a few other functions in file. I suggest you look for other functions in thre code that can be table-tested. We usually not treat this as an optional practice.

Reply:
added descriptions + unit tests for those (and other) functions, changed functions design to be table-tested.